pkg/utils/common.go

```diff
@@ -256,3 +258,20 @@ func ShouldPropagateNamespace(namespace string, skippedNamespaces map[string]boo
 	}
 	return true
 }
+
+// ExtractNumOfClustersFromPolicySnapshot extracts the numOfClusters value from the annotations
+// on a policy snapshot.
+func ExtractNumOfClustersFromPolicySnapshot(policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (int, error) {
```

Comment:
Yeah, putting the numOfClusters to the annotation is a compromise on many fronts. I remember the concern back then was that a) for obvious reasons we cannot include this field as part of the snapshot (this can be, as Ryan you have pointed out, avoided by excluding this field when computing the hash) b) it is a little bit weird to update a snapshot, which is supposed to be immutable.

Kubernetes does not have this concern as replicas is not part of the pod template; maybe we should move the placement type and numOfClusters out of policy? But that might be a little bit too much at this point 😣

Reply:
Hi Ryan! I will merge this PR first so as to unblock the next PR. If you would like to modify the numOfClusters related behavior, I will send a separate PR to address this.